### PR TITLE
Document the removal of forward class hints.

### PR DESCRIPTION
**Pull request by Zimmi48**
**Kind:** documentation

Follow-up of #13106.

### CODE REVIEW

**Review by ppedrot — COMMENTED** [review]
There is a bit of unrelated noise due to the renaming setoid_ring > ring, is is fine to leave it as a part of this PR?

**Comment by Zimmi48** [comment]
Yes, that's how we proceed until we enforce the update of the grammar files when the parser is changed.

**Comment by Zimmi48** [comment]
@gares There was a test case failure in async mode:

```
==========> TESTING success/ROmega2.v <==========
Segmentation fault (core dumped)
0m0.000000s 0m0.000000s
0m1.130000s 0m1.480000s
==========> FAILURE <==========
    success/ROmega2.v...Error! (should be accepted)
```

Maybe you should add `dmsg` calls to the test-suite job too?

File: doc/changelog/02-specification-language/13106-doc-and-changelog-for-13106.rst [hunk 1]
```diff
@@ -0,0 +1,5 @@
+- **Removed:**
+  Undocumented and experimental forward class hint feature ``:>>``.
+  Use ``:>`` (see :n:`@of_type`) instead
+  (`#13106 <https://github.com/coq/coq/pull/13106>`_,
+  by Pierre-Marie Pédrot).
```
Comment (jfehrle) [inline doc/changelog/02-specification-language/13106-doc-and-changelog-for-13106.rst:5]: @Zimmi48, shouldn't your name appear here?  And the PR number should be 13106?

**Review by jfehrle — APPROVED** [review]
I can merge after you respond.

File: doc/changelog/02-specification-language/13106-doc-and-changelog-for-13106.rst (hunk 1 above)
Comment (Zimmi48) [reply doc/changelog/02-specification-language/13106-doc-and-changelog-for-13106.rst:5]: That's right!

**Comment by jfehrle** [comment]
@coqbot: merge now

**Pull request merged**